subjects/graphql/README.md

```diff
@@ -0,0 +1,185 @@
+## GraphQL
+
+### Objectives
+
+The objective of this project is to learn graphql query language by creating profiles. It will be provided,\
+by the platform, a graphql endpoint that is connected to the database. So you can query this endpoint to obtain the information you desire.\
+**Note** that for security reasons some tables are private and some are public, you will only be provided with certain content.
+
+Your profile must have at least:
+
+- Basic user identification
+- XP amount
+- level
+- grades
+- skills
+- statistics
+
+### Instructions
+
+> Here is a small introduction and advantages of graphQL, if you want to read more about graphQL you can visit there [site](https://graphql.org/)
+
+GraphQL is a query language for any API endpoint and runtime. The syntax lets you specify what data you want to receive from that API.\
+You must have in mind that this language is used for API endpoints and not database. Unlike SQL,\
+graphql query does not go to your database directly but to your graphql API endpoint which can connect to a database or any other data source.
+
+You are already familiar with REST, since REST is a robust methodology of creating APIs and not elastic/scalable.\
```

Comment:
saying that rest is not scalable and graphQL is a bit bullshit talk, both can be scaled very well or very badly.

REST is very easy to cache, and this allow for very strong scalability if done properly.

`elastic` ? what does that mean ?

GraphQL strong point is an expressive query language in the client, this has a huge bunch of draw backs too, harder to limit / restrict and handle permissions.

Having a query language directly in the client then allow:
- retrieve only what you want
- batch queries in a single request
- move a lot of backend code to the client, as such helping with being sure query code match the actual client need and keeping the codebase clean.

I think you have 3 big wins:
- Front end teams don't need to ask back end teams to get what they want, so less friction for big teams
- Less data over the network, this is important for market with unreliable connections (3g phone, india, africa...)
- Speed of development, it's easier and faster to make queries in graphQL so you can move faster in your development.


Main trade of are:
- more code shipped too the front end side, heavier client applications, more complexity.
- much more ressource intensive for the servers, having to parse the query, and no caching make you do a lot more work
- as it is an abstraction that make it easier to query, you can easly do very expensive queries without them looking very complex, this is a hidden cost of the "fast & easy" queries,

Reply:
yha, i did try to say something like that but it seams that i failed. so i will just delete everything thats explaining graphql and the comparing to REST to graphql. Thank you for the points on graphql